packages/ui/src/common/phone-number-field/index.tsx

```diff
@@ -70,21 +62,29 @@ export const PhoneNumberField = React.forwardRef(function PhoneNumberField(
     const inputValue = e.target.value;
     if (inputValue.includes('+')) {
       setNumberAndIso(inputValue);
-      setSelectedCountry(countryOptions.find(c => c.iso === iso) || countryOptions[0]);
     } else {
       setNumber(inputValue);
     }
   };
 
-  React.useEffect(callOnChangeProp, [numberWithCode, onChange]);
-  React.useEffect(() => {
-    if (isOpen) {
-      commandInputRef.current?.focus();
-      setTimeout(() => {
-        commandListRef.current?.querySelector('[data-checked=true]')?.scrollIntoView({ block: 'start' });
-      }, 0);
-    }
-  }, [isOpen]);
+  React.useEffect(
+    function syncSelectedCountry() {
+      setSelectedCountry(countryOptions.find(c => c.iso === iso) || countryOptions[0]);
+    },
+    [iso],
+  );
```

Comment:
Ensures proper country is shown on load

https://linear.app/clerk/issue/SDKI-95/common-phone-number-field-isnt-automatically-localized